src/components/BannerConductor/StateMachine/states/ClosedState.ts

```diff
@@ -0,0 +1,19 @@
+import { BannerState } from '@src/components/BannerConductor/StateMachine/states/BannerState';
+import { BannerStates } from '@src/components/BannerConductor/StateMachine/BannerStates';
+
+export class ClosedState extends BannerState {
```

Comment:
The way the "Closed" state and state machine is set up currently, it'll probably encapsulate the tracking for "closing" the banner. Keep in mind that with the current setup we'll have different "events" that lead to a closed state:
- Size issue
- User interaction with the page (i.e. page object instance) before banner is displayed
- User interaction (with different close buttons / soft close, etc)

We could also split this state to make it more meaningful, into `Closed` (through direct user interaction with banner elements) and `Canceled` (page object prevented banner display). The `Canceled` state could then call the `mw.centralNotice.setBannerLoadedButHidden('reason')` method required by https://phabricator.wikimedia.org/T310982 
The `Canceled` constructor would also take the page object like the `Pending` one does, but we wouldn't need it for the `Closed` one, which is a good thing because the page object should be constrained to the `BannerConductor` (which is also the the one that makes the transition to `Canceled` 


The more I think of it, the more I like this state machine, because its constructor will force us to think about events when closing (not showing) the banner. 
We'll have to discuss how we handle the different tracking events (`BannerEvent` vs `SizeIssue`) and how (or if) we pass the event handler into the constructors. 

I'd like the API on the *calling* side to be as clear and lean as possible, which is why I'd go with some kind of factory method

```typescript
// examples of different events
bannerState.changeState(CloseState.newForBannerSizeIssue());
bannerState.changeState(CloseState.newForUserInteraction('soft-close-timeout'))
```

We'd need to figure out how we can share the event handling instance among the states without needing to pass it into every constructor. Maybe pass it into `BannerStateMachine` and pass it as a parameter to the `enter` and `exit` methods?